src/modules/decklink/producer/decklink_producer.cpp

```diff
@@ -382,10 +424,30 @@ class decklink_producer : public IDeckLinkInputCallback
     virtual ULONG STDMETHODCALLTYPE AddRef() { return 1; }
     virtual ULONG STDMETHODCALLTYPE Release() { return 1; }
 
-    virtual HRESULT STDMETHODCALLTYPE VideoInputFormatChanged(BMDVideoInputFormatChangedEvents /*notificationEvents*/,
-                                                              IDeckLinkDisplayMode* newDisplayMode,
+    virtual HRESULT STDMETHODCALLTYPE VideoInputFormatChanged(BMDVideoInputFormatChangedEvents notificationEvents,
+                                                              IDeckLinkDisplayMode*            newDisplayMode,
                                                               BMDDetectedVideoInputFormatFlags /*detectedSignalFlags*/)
     {
+        auto mode = newDisplayMode->GetDisplayMode();
+        auto fmt     = get_caspar_video_format(mode);
+
+        auto new_fmt     = core::video_format_desc(fmt);
+
+        CASPAR_LOG(info) << print() << L" Input format changed from " << input_format.name << L" to " << new_fmt.name;
+
+        input_format = new_fmt;
+
+        input_->PauseStreams();
```

Comment:
All of these input calls should be checked for success like we are in the constructor.
If any of these fail we will currently be ending up with a broken producer without anything in the logs.

It will need to be done in a try catch block like this so that the exception gets back to us correctly.
```
try {

} catch (...) {
    exception_ = std::current_exception();
    return E_FAIL;
}
```